libsolidity/codegen/CompilerContext.cpp

```diff
@@ -265,31 +265,38 @@ void CompilerContext::appendInlineAssembly(
 	}
 
 	unsigned startStackHeight = stackHeight();
-	auto identifierAccess = [&](
+
+	assembly::ExternalIdentifierAccess identifierAccess;
+	identifierAccess.resolve = [&](
+		assembly::Identifier const& _identifier,
+		assembly::IdentifierContext
+	) {
+		auto it = std::find(_localVariables.begin(), _localVariables.end(), _identifier.name);
+		return it == _localVariables.end() ? size_t(-1) : 1;
+	};
+	identifierAccess.generateCode = [&](
 		assembly::Identifier const& _identifier,
-		eth::Assembly& _assembly,
-		assembly::CodeGenerator::IdentifierContext _context
+		assembly::IdentifierContext _context,
+		eth::Assembly& _assembly
 	) {
 		auto it = std::find(_localVariables.begin(), _localVariables.end(), _identifier.name);
-		if (it == _localVariables.end())
-			return false;
+		solAssert(it != _localVariables.end(), "");
 		unsigned stackDepth = _localVariables.end() - it;
 		int stackDiff = _assembly.deposit() - startStackHeight + stackDepth;
```

Comment:
`_assembly.deposit()` is the current stack height as seen by the code generator. `startStackHeight` is the stack height (as seen by the code generator) at the start of the inline assembly block. `_localVariables` is a list of local variables that is passed to the inline assembly code generator that are assumed to be at the top of the stack at the point where the inline assembly block starts. So the number of slots we need to reach down is `currentHeight - initialHeight + depth_into_initial_local_variables`.